tests/test_wmb.py

```diff
@@ -87,10 +119,104 @@ def test_check_history(test_session):
         Station).filter(Station.native_id == native_id)
     assert q.count() == 2
 
+    # test error handle
+    err_lines = '''station_code,weather_date,precipitation
+1029,2012090312,12
+'''
+    err_data = []
+    f = StringIO(err_lines)
+    reader = csv.DictReader(f)
+    for row in reader:
+        err_data.append(row)
+
+    copy_hist = History(station_name='FIVE MILE')
+    test_session.add(copy_hist)
+    test_session.add(
+        Station(native_id='1029', network=o.network, histories=[copy_hist]))
+    o = ObsProcessor(test_session, err_data, 1000)
+    for row in o.data:
+        check = o.process_row(row)
+
+    assert check is None
+
 
 def test_process(test_session, test_data):
     o = ObsProcessor(test_session, test_data, 1000)
     o.process()
 
     q = test_session.query(Obs)
     assert q.count() == 8
+
+    # test UniquenessError handle
+    o.process()
+    assert o._obs_in_db == 5
+
+
+def test_parse_time_error_handle(test_session, test_data):
```

Comment:
Good, this test case is descriptive in its use of bad data and tests unique behaviuor.